src/osp/Active/SysNewton.cpp

```diff
@@ -557,33 +569,67 @@ std::pair<Matrix3, Vector4> SysNewton::compute_hier_inertia(ActiveScene& rScene,
     Vector4 centerOfMass =
         compute_hier_CoM<EIncludeRootMass::Include>(rScene, entity);
 
+    auto& reg = rScene.get_registry();
+
     // Sum inertias of children
-    for (ActiveEnt nextChild = rScene.reg_get<ACompHierarchy>(entity).m_childFirst;
+    for (ActiveEnt nextChild = reg.get<ACompHierarchy>(entity).m_childFirst;
         nextChild != entt::null;)
     {
-        // Compute the inertia of the child subhierarchy
-        auto const& [childInertia, childCoM] = compute_hier_inertia(rScene, nextChild);
-
         // Use child transformation to transform child inertia and add to sum
-        Matrix4 childTransform = rScene.reg_get<ACompTransform>(nextChild).m_transform;
-        Matrix3 rotation = childTransform.rotation();
-        // Offset is the vector between the ship center of mass and the child CoM
-        Vector3 offset = (childTransform.translation() + childCoM.xyz()) - centerOfMass.xyz();
-        I += phys::transform_inertia_tensor(
-            childInertia, childCoM.w(), offset, rotation);
-
-        nextChild = rScene.reg_get<ACompHierarchy>(nextChild).m_siblingNext;
+        auto const *childTransform = reg.try_get<ACompTransform>(nextChild);
+        Matrix4 const childTransformMat = (childTransform != nullptr)
+            ? childTransform->m_transform : Matrix4{Magnum::Math::IdentityInit};
+
+        /* Special case:
+         * Ordinarily, a child without a transformation means that branch of the
+         * hierarchy doesn't have any physics-related data and can be skipped.
+         * However, to minimize the number of transformations needed for
+         * entities like machines which are associated with an immediate parent
+         * and would have an identity transformation, the ACompTransform may be
+         * omitted. A child that has no transformation but still has a mass will
+         * be assumed to have an identity transformation relative to its parent
+         * and processed as usual.
+         */
+        if ((childTransform != nullptr) || reg.has<ACompMass>(nextChild))
```

Comment:
this `reg::has<>()` call will need to be turned into a `reg::all_of<ACompMass>(ent)` call when the PR is merged with master's updated EnTT version